content-security-policy/base-uri/base-uri_iframe_sandbox.sub.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+    <meta charset="utf-8">
+    <title>base-uri works correctly inside a sandboxed iframe.</title>
+    <script src='/resources/testharness.js'></script>
+    <script src='/resources/testharnessreport.js'></script>
+
+    <!-- CSP served: base-uri 'self' -->
+</head>
+
+<body>
+    <h1>base-uri works correctly inside a sandboxed iframe.</h1>
+    <div id='log'></div>
+
+    <script>
+        window.addEventListener('securitypolicyviolation', function(e) {
+            assert_unreached('No CSP violation report has fired.');
+        });
+
+        async_test(function(t) {
+            window.addEventListener('message', t.step_func(function(e) {
+                if (e.data.startsWith('SAME-ORIGIN,')) {
+                    assert_equals(e.data, 'SAME-ORIGIN,' + location.origin + '/base/');
+                    t.done();
+                }
+            }));
+            var e = document.createElement('iframe');
+            e.sandbox = 'allow-scripts';
+            e.style.display = 'none';
```

Comment:
Nit: Rather than doing this, did you consider just removing the frame after the `assert_equals` (e.g. `i.remove()`)?